### PR TITLE
Infra mvp agents

### PR DESCRIPTION
**Pull request by JaafarRammal**
# Summary

Setup polymorphism (?) by extending a base agent and spawn custom agents

## Additional Information

- Base agent has some private fields of interest with getters (e.g. UUID)
- Example with 2 custom agents in Team 1 folder where each has a different `Run` implementation and content 
- On the simulation level we know that all these custom agents are just `Agent` type and we abstract the logic at this level

## Test Plan

- [ ] Initialized agents have the expected HP and floor values (ensures no nasty override in `New`)
- [ ] All agents in the simulation env are in the tower and vice-versa (ensures no base agent override since UUID is only generated and assigned in the base agent `New`)

### CODE REVIEW

**Comment by jzzheng22** [comment]
Some of my changes include moving files around - I'll send you a diff on Slack in a bit.

File: pkg/agents/default/baseagent.go [hunk 1]
```diff
@@ -6,15 +6,21 @@ import (
 
 	tower "github.com/SOMAS2021/SOMAS2021/pkg/infra/tower"
```
Comment (jzzheng22) [inline pkg/agents/default/baseagent.go:7]:
```suggestion
	"github.com/SOMAS2021/SOMAS2021/pkg/infra/tower"
```

You will need to change the package name in tower to `tower`. 
i.e. the first line in `tower.go` should be `package tower`, not `package Tower`.
Capital letters in package names are bad practice in Go.

**Review by jzzheng22 — CHANGES_REQUESTED** [review]
I will look at this again in the morning but for now just this + the diff that I sent Jaafar.

File: pkg/infra/simulation/simulation.go [hunk 2]
```diff
@@ -3,7 +3,7 @@ package simulation
 import (
 	"log"
 
-	baseagent "github.com/SOMAS2021/SOMAS2021/pkg/agents/default"
+	"github.com/SOMAS2021/SOMAS2021/pkg/agents"
 	agent1 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team1/agent1"
 	agent2 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team1/agent2"
```
Comment (jzzheng22) [inline pkg/infra/simulation/simulation.go]:
```suggestion
	"github.com/SOMAS2021/SOMAS2021/pkg/agents/team1/agent1"
	"github.com/SOMAS2021/SOMAS2021/pkg/agents/team1/agent2"
```

If you can please change the first lines of `agent1.go` and `agent2.go` to:
`package agent1` and `package agent2` respectively <3
Package should be called whatever the folder name is called

**Review by jzzheng22 — CHANGES_REQUESTED** [review]

File: pkg/infra/simulation/simulation.go (hunk 2 above)
Comment (jzzheng22) [reply pkg/infra/simulation/simulation.go]: (and it means you don't need to rename the import)

**Review by udai-arneja — APPROVED** [review]
Looks good.

**Review by jzzheng22 — CHANGES_REQUESTED** [review]

**Review by jzzheng22 — APPROVED** [review]
LGTM. Thanks all

**Pull request merged**